src/parser/core/Combatant.ts

```diff
@@ -449,21 +449,38 @@ class Combatant extends Entity {
     return this._getGearItemBySlotId(GEAR_SLOTS.OFFHAND);
   }
 
+  private legendaries: Set<number> = new Set();
```

Comment:
happy to revert this if out of scope, `.hasLegendary` was previously just really ineffective and this now scans only once for both `.hasLegendary` and `.getItem`

Reply:
I'll let @emallson  handle this one :P

Reply:
This looks fine to me. Much more efficient to only compute it once.